assignments/closure.js

```diff
@@ -1,21 +1,57 @@
 // ==== Challenge 1: Write your own closure ====
 // Write a simple closure of your own creation.  Keep it simple!
 
+const hello1 = "Hello World!"
 
+function rHello () {
+	return hello1;
+}
+console.log(rHello())
 /* STRETCH PROBLEMS, Do not attempt until you have completed all previous tasks for today's project files */
 
 
 // ==== Challenge 2: Create a counter function ====
+var count = 0;
 const counter = () => {
-  // Return a function that when invoked increments and returns a counter variable.
-};
+  
+  return function() {
+    count = count + 1;
+    return count;
+  }
+}
+const newCounter = counter();
+console.log(newCounter())
+console.log(newCounter())
+console.log(newCounter())
 // Example usage: const newCounter = counter();
 // newCounter(); // 1
 // newCounter(); // 2
 
 // ==== Challenge 3: Create a counter function with an object that can increment and decrement ====
-const counterFactory = () => {
-  // Return an object that has two methods called `increment` and `decrement`.
-  // `increment` should increment a counter variable in closure scope and return it.
-  // `decrement` should decrement the counter variable and return it.
-};
+//let count = 0;
+var count = 0; // reseting count back to zero for challenge 3. That's why I used var count. I could have used let count and let count2, but meh. 
```

Comment:
Variables don't need to be in the global scope in order to create a closure. Indeed, most people use closures to prevent this kind of situation here. Lemme refactor challenge 2 to show this:

```js
const counter = () => {
    let count = 0;

    return function() {
        count = count + 1;
        return count;
    }
}

console.log(count); // count is not defined
```

Notice that the count variable is _inside_ the `counter` function. That means that count cannot be accessed in the global scope. But a closure still exists: the inner function can still reach out and get `count` in the surrounding `counter` function outside of it.

If you remove the `console.log(count)` and instead increment the counter:

```js
const countUp = counter();
console.log(countUp()); // 1
console.log(countUp()); // 2
console.log(countUp()); // 3
```

It still works as expected. If you have time, try to refactor Challenge 3 to be similar.